packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1013,8 +1015,11 @@ function commitNestedUnmounts(
 }
 
 function detachFiberMutation(fiber: Fiber) {
-  // Cut off the return pointers to disconnect it from the tree. Ideally, we
-  // should clear the child pointer of the parent alternate to let this
+  // Cut off the return pointers to disconnect it from the tree.
+  // Note that we can't clear child or sibling pointers yet,
+  // because they may be required for passive effects.
+  // These pointers will be cleared in a separate pass.
+  // Ideally, we should clear the child pointer of the parent alternate to let this
```

Comment:
If a Deletion always gets a passive effect scheduled on it, as suggested above, we could move this whole function to the passive phase. (Except maybe `return`  and `alternate`, since those affect semantics, like if you call `setState` on an unmounted component.)

Reply:
If we don't clear `return` during layout, it would break a few things:
* The unmounted state warning
* How events bubble for the new API Dominic has been working on

That being said, it would *fix* a limitation of the Profiler API when it comes to measuring time spent in passive effect cleanup. That time currently doesn't get reported during an unmount (either a regular unmount or because of an error boundary) because the `return` pointer has been detached previously and so it can't bubble up to the nearest Profiler node. If we deferred detaching it until after passive, that time would be reported which is a plus.

Not sure if there would be a viable workaround for the event case, but for the unmounted state warning- we *could* walk the `return` path and look for Fibers with `Deletion` effects scheduled, except that this would have false positives for deletions that were scheduled and not committed so that's a bummer.

Reply:
That being said, I believe we can safely defer everything *except* the `return` pointer cleanup to the passive effects phase which would be *slightly* better. I'll do that...